src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -54,6 +54,21 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
     sourceEntity.addEdge(label, targetVertex)
   }
 
+  def cloneAllEntities(entityDAO: EntityDAO, workspaceNamespace: String, newWorkspaceNamespace: String, workspaceName: String, newWorkspaceName: String, txn: RawlsTransaction): Unit = txn withGraph { db =>
+    val workspace = getWorkspaceVertex(db, newWorkspaceNamespace, newWorkspaceName)
+      .getOrElse(throw new IllegalArgumentException("Cannot clone entity to nonexistent workspace " + workspaceNamespace + "::" + workspaceName))
+
+    //copy the entities
+    listEntitiesAllTypes(workspaceNamespace, workspaceName, txn).foreach { entity =>
+      db.addVertex(entity)
```

Comment:
I think we want db.addVertex(null), then set properties of the vertex - this is passing entity as the id parameter, and I'm not sure what that will end up doing